app/data/tests/test_views.py

```diff
@@ -123,3 +125,31 @@ def test_tilekey_param(self):
             force_authenticate(request, user=self.admin)
             response = view(request)
             self.assertEqual(mocked_redis.call_count, 1)
+
+    def test_get_serializer_class(self):
+        """Test that get_serializer_class returns read-only serializer correctly"""
+        with mock.patch('data.views.is_admin_or_writer') as views_not_public:
+            views_not_public.return_value = False
+            view = DriverRecordViewSet()
+            mock_req = mock.Mock(spec=Request)
+            mock_req.user = self.admin
+            view.request = mock_req
+            # Even though the user is an admin, we've patched is_admin_or_writer to return False
```

Comment:
Good call; I had thought I could get away with mocking just the one function, but that didn't work out so it ended up kind of a mishmash. I'll clean that up.